sympy/core/relational.py

```diff
@@ -203,6 +203,44 @@ def __nonzero__(self):
 
     __bool__ = __nonzero__
 
+    def as_solution_set(self, gen=False):
+        """
+        Rewrites the simplified (solution) inequality in terms of real set
+
+        Examples
+        ========
+
+        >>> from sympy import Symbol, Eq
+        >>> x = Symbol('x', real=True)
+        >>> (x>0).as_solution_set()
+        (0, oo)
+        >>> Eq(x, 0).as_solution_set()
+        {0}
+        >>> Ge(y, x).as_solution_set(y)
+        [x, oo)
+        >>> Lt(y, x).as_solution_set(y)
+        (-oo, x)
+        """
+        from sympy.solvers.inequalities import solve_univariate_inequality
+        syms = self.free_symbols
+
+        if len(syms) == 1:
+            sym = syms.pop()
+            return solve_univariate_inequality(self, sym, relational=False)
+        else:
+            from sympy import Interval
+            e = self.lhs - self.rhs
+            coeff_sign = e.coeff(gen)
```

Comment:
Yeah, we can have any kind of expression in an inequality. I will explain with an example:

In: a = And(x > 1, x + y < 1)
In: a.as_set(x, y)

Now this expression will be solved by inequalities solver (I will make a call to multivariate inequality solver in place where it currently raises NotImplementedError). Internally in that solver it will reach to result = And(x > 1, y < 1 - x). So how will it convert this result into ProductSet?. It will now use as_solution_set to convert the solution into ProductSet (Now we can have some assumptions in the solution inequalities like either lhs or rhs will have a Symbol). So basically these methods are for internal use of solvers only (Should I apply a leading underscore?).
So whenever we need to convert these logic expressions into sets then as_set will be called.